frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -238,29 +243,660 @@ def operation(
+                bounds of smaller
+
+            smaller_is_lhs (bool):
+                whether smaller is lhs of the comparison
+
+            smaller_is_rhs (bool):
+                whether smaller is rhs of the comparison
+        """
+
+        assert smaller.original_bit_width != bigger.original_bit_width
+
+        intermediate_bit_width = max(
+            smaller.bit_width,
+            min(
+                smaller_minus_clipped_bigger_dtype.bit_width,
+                clipped_bigger_minus_smaller_dtype.bit_width,
+            ),
+        )
+        intermediate_scalar_type = self.esint(intermediate_bit_width)
+
+        if smaller.bit_width != intermediate_bit_width:
+            smaller = self.cast(self.tensor(intermediate_scalar_type, smaller.shape), smaller)
+
+        low_bound = smaller_bounds[0] - 1
+        high_bound = smaller_bounds[1] + 1
+
+        clipper_lut = []
+        if bigger.is_unsigned:
+            clipper_lut += [
+                np.clip(i, low_bound, high_bound) for i in range(0, 2**bigger.bit_width)
+            ]
+        else:
+            clipper_lut += [
+                np.clip(i, low_bound, high_bound) for i in range(0, 2 ** (bigger.bit_width - 1))
+            ]
+            clipper_lut += [
+                np.clip(i, low_bound, high_bound) for i in range(-(2 ** (bigger.bit_width - 1)), 0)
+            ]
+        clipped_bigger = self.tlu(
+            self.tensor(intermediate_scalar_type, bigger.shape),
+            bigger,
+            clipper_lut,
+        )
+
+        invert_accept = False
+
+        do_clipped_bigger_minus_smaller = (
+            clipped_bigger_minus_smaller_dtype.bit_width
+            <= smaller_minus_clipped_bigger_dtype.bit_width
+        )
+        if do_clipped_bigger_minus_smaller:
+            lhs = clipped_bigger
+            rhs = smaller
+            lhs_minus_rhs_dtype = clipped_bigger_minus_smaller_dtype
+            if smaller_is_lhs:
+                invert_accept = True
+        else:
+            lhs = smaller
+            rhs = clipped_bigger
+            lhs_minus_rhs_dtype = smaller_minus_clipped_bigger_dtype
+            if smaller_is_rhs:
+                invert_accept = True
+
+        if invert_accept:
+            new_accept = set()
+            if Comparison.EQUAL in accept:
+                new_accept.add(Comparison.EQUAL)
+            if Comparison.LESS in accept:
+                new_accept.add(Comparison.GREATER)
+            if Comparison.GREATER in accept:
+                new_accept.add(Comparison.LESS)
+            accept = new_accept
+
+        intermediate_type = self.tensor(intermediate_scalar_type, resulting_type.shape)
+        lhs_minus_rhs = self.sub(intermediate_type, lhs, rhs)
+
+        accept_equal = int(Comparison.EQUAL in accept)
+        accept_greater = int(Comparison.GREATER in accept)
+        accept_less = int(Comparison.LESS in accept)
+
+        all_cells = 2**intermediate_bit_width
```

Comment:
this is 100% common with comparison_with_subtraction_trick
def comparison_on_substraction